Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -4,103 +4,156 @@
 using System.IO;
 using System.Linq;
 using UnityEditor;
-using UnityEngine.InputSystem.Utilities;
 
 namespace UnityEngine.InputSystem.Editor
 {
     internal static class ProjectWideActionsAsset
     {
-        internal const string kDefaultAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.inputactions";
-        internal const string kAssetPath = "ProjectSettings/InputManager.asset";
-        internal const string kAssetName = InputSystem.kProjectWideActionsAssetName;
+        internal const string kTemplateAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.inputactions";
+        internal const string kInputManagerAssetPath = "ProjectSettings/InputManager.asset";
+        internal const string kAssetPath = "ProjectSettings/InputSystemActions.inputactions";
 
-        static string s_DefaultAssetPath = kDefaultAssetPath;
-        static string s_AssetPath = kAssetPath;
+        static InputActionAsset s_TestAsset = null;
+
+        const string kAdditionalFilePath = "Assets/actions.InputSystemActionsAPIGenerator.additionalfile"; // Copy of asset that is fed to the SourceGenerator
 
 #if UNITY_INCLUDE_TESTS
-        internal static void SetAssetPaths(string defaultAssetPath, string assetPath)
+        internal static void SetTestAsset(InputActionAsset testAsset)
         {
-            s_DefaultAssetPath = defaultAssetPath;
-            s_AssetPath = assetPath;
+            s_TestAsset = testAsset;
         }
 
-        internal static void Reset()
+        internal static void ResetTestAsset()
         {
-            s_DefaultAssetPath = kDefaultAssetPath;
-            s_AssetPath = kAssetPath;
+            s_TestAsset = null;
         }
 
 #endif
 
+        // Ensures the project-wide settings are initialized early.
         [InitializeOnLoadMethod]
         internal static void InstallProjectWideActions()
         {
             GetOrCreate();
         }
 
-        internal static InputActionAsset GetOrCreate()
+        // Copies the inputactionasset content from the Project Setting asset into a file that
+        // the Roslyn source generator will be able to read.
+        internal static void CreateRoslynAdditionalFileForAsset()
+        {
+            try
+            {
+                if (File.Exists(kAssetPath))
+                {
+                    File.Copy(kAssetPath, kAdditionalFilePath, true);
```

Comment:
Seems we are stuck with a fixed path here too, would be nice if we could still allow a user to move this. Perhaps querying the asset DB for a file of the specific name prior to saving so we overwrite a moved version rather than just recreating